vlsv_reader_parallel.cpp

```diff
@@ -352,18 +556,78 @@ namespace vlsv {
     * @return If true, this process read its data successfully.*/
    bool ParallelReader::readArray(const std::string& tagName,const std::list<std::pair<std::string,std::string> >& attribs,
                                   const uint64_t& begin,const uint64_t& amount,char* buffer) {
-      if (parallelFileOpen == false) return false;
       bool success = true;
 
       // Fetch array info to all processes:
       if (getArrayInfo(tagName,attribs) == false) return false;
       const MPI_Offset start = arrayOpen.offset + begin*arrayOpen.vectorSize*arrayOpen.dataSize;
-      const int readBytes    = amount*arrayOpen.vectorSize*arrayOpen.dataSize;
+      const size_t readBytes = amount*arrayOpen.vectorSize*arrayOpen.dataSize;
+
+      // If readBytes is larger than getMaxBytesPerRead() this process needs 
+      // more than one collective call to read in all the data.
+      const size_t maxBytes = getMaxBytesPerRead();
+      size_t myExtraCollectiveReads = readBytes / maxBytes;
+
+      // There's always at least one read:
+      ++myExtraCollectiveReads;
```

Comment:
This is not needed if readBytes = n * maxBytes. In this (uncommon) case readBytes / maxBytes = n reads would be enough.